saplings/circuits/src/components/forms/ProposeCircuitForm.js

```diff
@@ -0,0 +1,263 @@
+    }
+    default:
+      throw new Error(`unhandled action type: ${action.type}`);
+  }
+};
+
+export function ProposeCircuitForm() {
+  const allNodes = useNodeRegistryState();
+  const localNodeID = useLocalNodeState();
+  const [modalActive, setModalActive] = useState(false);
+  const [localNode] = allNodes.filter(node => node.identity === localNodeID);
+  const [nodesState, nodesDispatcher] = useReducer(nodesReducer, {
+    selectedNodes: [],
+    availableNodes: [],
+    filteredNodes: {
+      nodes: [],
+      filteredBy: ''
+    },
+    error: ''
+  });
+
+  const nodesAreValid = () => {
+    return nodesState.selectedNodes.length >= 2;
+  };
+
+  const plusSign = (
+    <span className="add-sign">
+      <FontAwesomeIcon icon="plus" />
+    </span>
+  );
+
+  useEffect(() => {
+    if (allNodes) {
+      nodesDispatcher({
+        type: 'set',
+        nodes: allNodes
+      });
+    }
+  }, [allNodes]);
+
+  useEffect(() => {
+    if (localNode) {
+      nodesDispatcher({
+        type: 'select',
+        node: localNode
+      });
+    }
+  }, [localNode]);
+
+  return (
+    <MultiStepForm
+      formName="Propose Circuit"
+      handleSubmit={() => {}}
+      disabled={!nodesAreValid()}
+    >
+      <Step step={1} label="Add Nodes">
+        <div className="node-registry-wrapper">
+          <div className="selected-nodes-header">
+            <div className="title">Selected nodes</div>
+          </div>
+          <div className="form-error">{nodesState.error}</div>
+          <div className="selected-nodes">
+            <Chips>
+              {nodesState.selectedNodes.map(node => {
+                const local = node.identity === localNodeID;
+                return (
+                  <Chip
+                    node={node}
+                    isLocal={local}
+                    deleteable={!local}
+                    removeFn={() => {
+                      nodesDispatcher({ type: 'removeSelect', node });
+                    }}
+                  />
+                );
+              })}
+            </Chips>
+          </div>
+          <div className="available-nodes">
+            <div className="available-nodes-header">
+              <div className="title-wrapper">
+                <div className="title">Available nodes</div>
+                <input
+                  type="text"
+                  placeholder="Filter"
+                  className="search-nodes-input"
+                  onKeyUp={event => {
+                    nodesDispatcher({
+                      type: 'filter',
+                      input: event.target.value.toLowerCase()
+                    });
+                  }}
+                />
+              </div>
+              <button
+                type="button"
+                className="new-node-button"
+                onClick={() => setModalActive(true)}
+              >
+                {plusSign}
+                New node
+              </button>
+            </div>
+            <ul>
```

Comment:
I don't know if I like that the nodes disappear from the table when selected. Maybe just add a darker background or an icon?

Reply:
updated